src/mixins/raster-layer-line-mixin.js

```diff
@@ -0,0 +1,869 @@
+    scales.push({
+      name: getSizeScaleName(layerName),
+      type: "linear",
+      domain: (size.domain === "auto" ? {data: xformDataSource, fields: scaleDomainFields.size} : size.domain),
+      range: size.range,
+      clamp: true
+    })
+  }
+
+  if (typeof color === "object" && color.type === "ordinal") {
+    scales.push({
+      name: getColorScaleName(layerName),
+      type: "ordinal",
+      domain: (color.domain === "auto" ? {data: xformDataSource, fields: scaleDomainFields.color} : color.domain),
+      range: color.range,
+      default: color.range[color.range.length - 1],
+      nullValue: "#CACACA"
+    })
+  }
+
+  if (typeof color === "object" && color.type === "quantitative") {
+    scales.push({
+      name: getColorScaleName(layerName),
+      type: "quantize",
+      domain: (color.domain === "auto" ? {data: xformDataSource, fields: scaleDomainFields.color} : color.domain),
+      range: color.range,
+      nullValue: "#CACACA"
+    })
+  }
+
+  return scales
+}
+
+export default function rasterLayerLineMixin(_layer) {
+  let state = null
+  _layer.colorDomain = createRasterLayerGetterSetter(_layer, null)
+  _layer.sizeDomain = createRasterLayerGetterSetter(_layer, null)
+
+  _layer.setState = function(setter) {
+    if (typeof setter === "function") {
+      state = setter(state)
+    } else {
+      state = setter
+    }
+
+    if (!state.hasOwnProperty("transform")) {
+      state.transform = {}
+    }
+
+    return _layer
+  }
+
+  _layer.getState = function() {
+    return state
+  }
+
+  _layer.getProjections = function() {
+    return getTransforms(
+      "",
+      "",
+      "",
+      state,
+      lastFilteredSize(_layer.crossfilter().getId())
+    )
+      .filter(
+        transform =>
+          transform.type === "project" && transform.hasOwnProperty("as")
+      )
+      .map(projection => parser.parseTransform({ select: [] }, projection))
+      .map(sql => sql.select[0])
+  }
+
+  function getAutoColorVegaTransforms(aggregateNode) {
+    const rtnobj = {transforms: [], fields: []}
+    if (state.encoding.color.type === "quantitative") {
+      const minoutput = "mincolor", maxoutput = "maxcolor"
+      aggregateNode.fields = aggregateNode.fields.concat(["strokeColor", "strokeColor", "strokeColor", "strokeColor"])
+      aggregateNode.ops = aggregateNode.ops.concat(["min", "max", "avg", "stddev"])
+      aggregateNode.as = aggregateNode.as.concat(["mincol", "maxcol", "avgcol", "stdcol"])
+      rtnobj.transforms.push(
+        {
+          type: "formula",
+          expr: "max(mincol, avgcol-2*stdcol)",
+          as: minoutput
+        },
+        {
+          type: "formula",
+          expr: "min(maxcol, avgcol+2*stdcol)",
+          as: maxoutput
+        }
+      )
+      rtnobj.fields = [minoutput, maxoutput]
+    } else if (state.encoding.color.type === "ordinal") {
+      const output = "distinctcolor"
+      aggregateNode.fields.push("color")
+      aggregateNode.ops.push("distinct")
```

Comment:
We can revisit when we support `auto` for `ordinal` color type. Right now, this block of code will not be used because we only support `auto` for  `quantitative` type. This part is basically copied from point.

Reply:
oh, right. which means I probably added it to points as a placeholder -- just an FYI then.